connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/DelegatingClassLoaderTest.java

```diff
@@ -60,23 +70,33 @@ public class DelegatingClassLoaderTest {
 
     @BeforeEach
     @SuppressWarnings({"unchecked"})
-    public void setUp() {
+    public void setUp() throws InvalidVersionSpecificationException {
+        range = VersionRange.createFromVersionSpec("[" + version + "]");
         parent = mock(PluginClassLoader.class);
         pluginLoader = mock(PluginClassLoader.class);
+        pluginLoader2 = mock(PluginClassLoader.class);
         classLoader = new DelegatingClassLoader(parent);
         SortedSet<PluginDesc<SinkConnector>> sinkConnectors = new TreeSet<>();
         // Lie to the DCL that this arbitrary class is a connector, since all real connector classes we have access to
         // are forced to be non-isolated by PluginUtils.shouldLoadInIsolation.
         when(pluginLoader.location()).thenReturn("some-location");
-        pluginDesc = new PluginDesc<>((Class<? extends SinkConnector>) ARBITRARY_CLASS, null, PluginType.SINK, pluginLoader);
-        assertTrue(PluginUtils.shouldLoadInIsolation(pluginDesc.className()));
-        sinkConnectors.add(pluginDesc);
+        when(pluginLoader2.location()).thenReturn("other-location");
+        connectorPluginDesc = new PluginDesc<>((Class<? extends SinkConnector>) ARBITRARY_CLASS, null, PluginType.SINK, pluginLoader);
+        assertTrue(PluginUtils.shouldLoadInIsolation(connectorPluginDesc.className()));
+        sinkConnectors.add(connectorPluginDesc);
+        SortedSet<PluginDesc<Transformation<?>>> transformations = new TreeSet<>();
+        transformationPluginDesc = new PluginDesc<>((Class<? extends Transformation<?>>) ARBITRARY_CLASS, null, PluginType.TRANSFORMATION, pluginLoader);
+        transformationPluginDesc2 = new PluginDesc<>((Class<? extends Transformation<?>>) ARBITRARY_CLASS, null, PluginType.TRANSFORMATION, pluginLoader2);
+        transformationPluginDesc3 = new PluginDesc<>((Class<? extends Transformation<?>>) ARBITRARY_CLASS, version, PluginType.TRANSFORMATION, pluginLoader);
```

Comment:
nit: having a plugindesc with the same class but different version in the same plugin loader is physically impossible/an illegal state.